src/project.lua

```diff
@@ -125,23 +140,12 @@ function Project:excludeFiles()
       "^"..utils.lua.escape_string_regex(self.projectDirectory).."/",
       "")
   if rm_dir > 0 then
-    rm_dir = true
     dir = "^"..dir
-  else
-    rm_dir = false
   end
 
-  if rm_dir then
-    local rm = false
-    local file
-    local n = #self._fileList
-    for i = 1, n do
-      file = self._fileList[i]
-      if rm_dir then if file:find(dir) then rm = true end end
-      if rm then
-        self._fileList[i] = nil
-        rm = false
-      end
+  for i=#self._fileList,1,-1 do
+    if isExcluded(self._fileList[i], dir, unpack(self.excludeFileList)) then
```

Comment:
Avoid the second one, you should never modify Lua's built-in modules since that is not expected from a module and can end up with problems in other modules later on